src/uwtools/drivers/forecast.py

```diff
@@ -151,40 +178,15 @@ def run(self, cycle: datetime) -> bool:
 
         :return: Did the FV3 run exit with success status?
         """
-        # Prepare directories.
-        run_directory = self._config["run_dir"]
-        self.create_directory_structure(run_directory, "delete")
-
-        self._prepare_config_files(Path(run_directory))
-
-        self._config["cycle-dependent"].update(self._define_boundary_files())
-
-        for file_category in ["static", "cycle-dependent"]:
-            self.stage_files(run_directory, self._config[file_category], link_files=True)
-
-        if self._batch_script is not None:
-            batch_script = self.batch_script()
-            outpath = Path(run_directory) / self._batch_script
-
-            if self._dry_run:
-                # Apply switch to allow user to view the run command of config.
-                # This will not run the job.
-                log.info("Batch Script:")
-                batch_script.dump(None)
-                return True
-
-            batch_script.dump(outpath)
-            return self.scheduler.submit_job(outpath)
-
-        pre_run = self._mpi_env_variables(" ")
-        full_cmd = f"{pre_run} {self.run_cmd()}"
+        if self._batch_script:
+            method = self._run_via_batch_submission()
+        else:
+            method = self._run_via_local_execution()
```

Comment:
`self._run_via_local_execution()` is an actual _call_ of the method, rather than just a reference to the method itself. So the value of that expression is not a "method", but whatever it is that that function returns. It's ok to do it this way, but then the LHS should probably not be `method` but `status, output`. Or it could be, say, `result` and then `status, output = method` would change to `status, output = result`.

But how about the more concise
```
method = self._run_via_batch_submission if self._batch_script else self._run_via_local_execution
status, output = method()
```
This sets the variable `method` to point to one of the two private methods, and then the expression `method()` performs the call and assigns the results of the call to the `status` and `output` variables.

Or we could have the single line
```
status, output = self._run_via_batch_submission() if self._batch_script else self._run_via_local_execution()
```
`black` will reformat that, since it's too long for one line, but that one statement should do.